mutiny-core/src/nodemanager.rs

```diff
@@ -142,13 +142,22 @@ pub struct MutinyChannel {
 
 impl From<&ChannelDetails> for MutinyChannel {
     fn from(c: &ChannelDetails) -> Self {
+        let size = c.channel_value_satoshis;
+        let balance = c.next_outbound_htlc_limit_msat / 1_000;
+        let inbound = c.inbound_capacity_msat / 1_000;
+        let mut reserve = ((c.outbound_capacity_msat - c.next_outbound_htlc_limit_msat) / 1_000)
+            + c.unspendable_punishment_reserve.unwrap_or(0);
+
+        // if there is any diff in the channel size and the other totals,
+        // just add it to the reserve to avoid confusion
+        reserve += size - (balance + inbound + reserve);
```

Comment:
I don't understand this. You're adding to reserve an amount that subtracts the reserve. I'd like to see some numbers / tests for these calculations.

Reply:
Okay simplified